### PR TITLE
feat: move library to typescript code generation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ docs/
 .DS_Store
 package-lock.json
 __pycache__
+build/
```
Comment (alexander-fenster) [inline .gitignore]: new line :)

**Comment by xiaozhenliu-gg5** [comment]
Waiting for a grpc config change from //google3 and //third_party, timeout needs to be changed for passing samples-test.  Will regenerate the client with the latest change from API.

**Comment by xiaozhenliu-gg5** [comment]
cl/286294394 submitted, rerun synthtool to pick up the changes, and replace `initial_rpc_timeout_millis ` and `max_rpc_timeout_millis ` timeout in synth.py. Sample-tests pass now.

File: .gitignore (hunk 1 above)
Comment (alexander-fenster) [reply .gitignore]: new line still not here :)

File: samples/quickstart.js [hunk 2]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 'use strict';
+/*eslint no-warning-comments: [0, { "terms": ["todo", "fixme"], "location": "anywhere" }]*/
```
Comment (alexander-fenster) [inline samples/quickstart.js]: do we really need this? (here and in other files)

File: webpack.config.js [hunk 3]
```diff
@@ -12,33 +12,51 @@
...
 
+const path = require('path');
+
 module.exports = {
-  entry: './src/browser.js',
+  entry: './src/index.ts',
   output: {
-    library: 'tasks',
-    filename: './tasks.js',
+    library: 'CloudTasks',
```
Comment (alexander-fenster) [inline webpack.config.js]: breaking change, need to use `--main-service tasks`

File: .gitignore (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply .gitignore]: sorry! missed the comments here

File: samples/quickstart.js (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply samples/quickstart.js]: `npm run lint` complained if deleted this line
```
warning  Unexpected 'todo' comment  no-warning-comments
```

File: webpack.config.js (hunk 3 above)
Comment (xiaozhenliu-gg5) [reply webpack.config.js]: got it! thanks

**Pull request merged**